packages/app/src/domain/vaccine/components/vaccine-delivery-and-administrations-tooltip.tsx

```diff
@@ -0,0 +1,152 @@
+  return input.type === 'line';
+}
+
+function isStackedConfig(
+  input: SeriesConfig<VaccineDeliveryAndAdministrationsValue>[number]
+): input is StackedAreaSeriesDefinition<VaccineDeliveryAndAdministrationsValue> {
+  return input.type === 'stacked-area';
+}
+
+export function VaccineDeliveryAndAdministrationsTooltip({
+  data,
+  estimateRange,
+}: {
+  data: TooltipData<VaccineDeliveryAndAdministrationsValue>;
+  estimateRange: [number, number];
+}) {
+  const { siteText, formatNumber, formatDateFromSeconds } = useIntl();
+
+  const isEstimate =
+    data.value.date_start_unix >= estimateRange[0] &&
+    data.value.date_end_unix <= estimateRange[1];
+
+  const firstValue = data.value.total_delivered;
+  const firstConfig = data.config
+    .filter(isLineConfig)
+    .find((x) => x.metricProperty === 'total_delivered');
+
+  const otherConfigs = data.config.filter(isStackedConfig);
+
+  if (!firstConfig || !otherConfigs.length) {
+    return null;
+  }
+
+  const dateEndString = formatDateFromSeconds(
+    data.value.date_end_unix,
+    'day-month'
+  );
+
+  return (
+    <>
+      <VisuallyHidden>{dateEndString}</VisuallyHidden>
+      <TooltipList>
+        <TooltipListItem>
+          <span>
+            <ColorIndicator color={firstConfig.color} />
+            {isEstimate
+              ? siteText.vaccinaties.data.vaccination_chart.estimated
+              : siteText.vaccinaties.data.vaccination_chart.delivered}
+            :
+          </span>
+          <TooltipValueContainer>
+            {formatNumber(firstValue)}
+          </TooltipValueContainer>
+        </TooltipListItem>
+      </TooltipList>
+
+      <Heading level={5} my={1}>
+        {!isEstimate
+          ? siteText.vaccinaties.data.vaccination_chart.doses_administered
+          : siteText.vaccinaties.data.vaccination_chart
+              .doses_administered_estimated}
+      </Heading>
+
+      <TooltipList>
+        {otherConfigs.map((config) => (
+          <TooltipListItem key={config.label}>
+            <span>
+              <ColorIndicator color={config.color} />
+              {formatLabel(config.shortLabel ?? config.label, siteText)}:
+            </span>
+            <TooltipValueContainer>
+              <strong>
+                {formatNumber(data.value[config.metricProperty] || 0)}
+              </strong>
+            </TooltipValueContainer>
+          </TooltipListItem>
+        ))}
+
+        <Spacer mb={1} />
+        <TooltipListItem>
+          <span>
+            <ColorIndicator color="transparent" />
+            {
+              siteText.vaccinaties.data.vaccination_chart
+                .doses_administered_total
+            }
+            :
+          </span>
+          <TooltipValueContainer>
+            <strong>{formatNumber(data.value.total)}</strong>
+          </TooltipValueContainer>
+        </TooltipListItem>
+      </TooltipList>
+    </>
+  );
+}
+
+function formatLabel(labelKey: string | undefined, text: SiteText) {
```

Comment:
Optional arguments typically come last, and then you con't need the | undefined. 

```suggestion
function formatLabel(text: SiteText, labelKey?: string) {
```